### PR TITLE
ref(code-snippet): Introduce the prop 'disableUserSelection'

### PR DESCRIPTION
**Pull request by priscilawebdev**
~This PR adds a new prop called "loading" to the code snippet component. This change ensures that when specific parts of the code snippet are still being fetched asynchronously, users won't be able to copy the incomplete code. The entire code snippet will also be blocked until all the necessary parts are ready. This change is being used in the https://github.com/getsentry/sentry/pull/54675. (there is a video showing case how the feature works)~

This PR adds a new prop called "disableUserSelection" to the code snippet component. This change ensures that when specific parts of the code snippet are still being fetched asynchronously, users won't be able to copy the incomplete code manually.

### CODE REVIEW

**Review by evanpurkhiser — COMMENTED** [review]
Will leave this one to @vuluongj20 to review

**Comment by vuluongj20** [comment]
Hmm why should this be a `<CodeSnippet />` prop? Couldn't we just render `<Placeholder />`/`<LoadingIndicator />` wherever the code is being loaded, and replace them with `<CodeSnippet />` when the code is ready? The way it's implemented, when `loading` is true `<CodeSnippet />` will just look like a `<Placeholder />` with no content inside.

**Comment by priscilawebdev** [comment]
> Hmm why should this be a <CodeSnippet /> prop? Couldn't we just render <Placeholder />/<LoadingIndicator /> wherever the code is being loaded, and replace them with <CodeSnippet /> when the code is ready? The way it's implemented, when loading is true <CodeSnippet /> will just look like a <Placeholder /> with no content inside.

Yes, we considered this option, but there was a "jump" effect when trying it. To make it work well, we would need to know the code snippet's size beforehand. Since layout calculations are "expensive" and the fetch shouldn't take long, this was a quick win solution

File: static/app/components/codeSnippet.tsx [hunk 1]
```diff
@@ -13,6 +13,12 @@ interface CodeSnippetProps {
   language: string;
   className?: string;
   dark?: boolean;
+  /**
+   * Makes the text of the element and its sub-elements not selectable.
+   * Userful when loading parts of a code snippet, and
+   * we wish to avoid users copying them manually.
+   */
+  disabledUserSelection?: boolean;
```
Comment (vuluongj20) [inline static/app/components/codeSnippet.tsx:21]: let's call this `disableUserSelection`, without the `d`

Comment (priscilawebdev) [reply static/app/components/codeSnippet.tsx]: done

**Review by vuluongj20 — APPROVED** [review]

**Pull request merged**